docs/cloud/azureai/use-flow-in-azure-ml-pipeline.md

```diff
@@ -118,23 +271,69 @@ def pipeline_func_with_flow(data):
                 "deployment_name": "text-davinci-003",
             },
         },
+        environment_variables={
+            "AZURE_OPENAI_API_KEY": "${my_connection.api_key}",
+            "AZURE_OPENAI_API_BASE": "${my_connection.api_base}",
+        }
     )
     flow_node.compute = "cpu-cluster"
+    flow_node.instance_count = 2
+    flow_node.max_concurrency_per_instance = 2
+    flow_node.mini_batch_error_threshold = 5
+    flow_node.retry_settings = RetrySettings(timeout=30, max_retries=5)
 
 pipeline_with_flow = pipeline_func_with_flow(data=data_input)
 
 pipeline_job = ml_client.jobs.create_or_update(pipeline_with_flow)
 ml_client.jobs.stream(pipeline_job.name)
 ```
 
-Above is part of the pipeline job python code, see here for [full example](https://github.com/Azure/azureml-examples/tree/main/sdk/python/jobs/pipelines/1l_flow_in_pipeline).
-
 :::
 
 ::::
 
+## Environment of the component
+
+By default, the environment of the created component will be based on the latest promptflow runtime image. If customer has [specified python requirement file](../../reference/flow-yaml-schema-reference.md) in `flow.dag.yaml`, they will be applied to the environment automatically:
+
+``` yaml
+...
+environment:
+  python_requirements_txt: requirements.txt
+```
+
+If customer want to use an existing Azure ML environment or define the environment in Azure ML style, they can define it in `run.yaml` like below:
+
+```yaml
+$schema: https://azuremlschemas.azureedge.net/promptflow/latest/Run.schema.json
+flow: <my-flow-directory>
+azureml:
+  environment: azureml:my-environment:1
+```
+
+For more details about the supported format of Azure ML environment, please refer to [this doc](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-manage-environments-v2?view=azureml-api-2&tabs=cli).
+
 ## Difference across flow in prompt flow and pipeline job
 
 In prompt flow, flow runs on [runtime](https://learn.microsoft.com/en-us/azure/machine-learning/prompt-flow/concept-runtime), which is designed for prompt flow and easy to customize; while in pipeline job, flow runs on different types of compute, and usually compute cluster.
 
 Given above, if your flow has logic relying on identity or environment variable, please be aware of this difference as you might run into some unexpected error(s) when the flow runs in pipeline job, and you might need some extra configurations to make it work.
+
+## Component ports and run settings
+
```

Comment:
This table looks like
![image](https://github.com/microsoft/promptflow/assets/2208599/328a650a-4fff-4cd1-86db-a8913818f0e6)

Do you want to split these  "input ports" to some sub items and split this big table to some smaller ones?

Reply:
updated, please help check if it fits your expectation